R/install.R

```diff
@@ -0,0 +1,264 @@
+        "",
+        "  brew install python",
+        "",
+        "If Homebrew + Python are installed and problems persist, try reinstalling:",
+        "",
+        "  brew reinstall python",
+        "",
+        "After installation, make sure that Homebrew is on your PATH env var.",
+        "If problems persist, set RETICULATE_PYTHON manually and re-install:",
+        "",
+        "  Sys.setenv(RETICULATE_PYTHON = \"/path/to/your/python\")",
+        "  toolchest::install_toolchest()",
+        sep = "\n"
+      )
+      packageStartupMessage(suggestion_message)
+    } else if (sysname == "Windows") {
+      # If on Windows, attempt to install Miniconda if not installed.
+      conda_try <- try(reticulate::conda_list(), silent = TRUE)
+      if (class(conda_try) == "try-error") {
+        packageStartupMessage("No install of Anaconda detected. Installing Miniconda.")
+        reticulate::conda_install(
+          "r-miniconda",
+          python_version = PYTHON_VERSION_TO_BE_INSTALLED
+        )
+        return(reticulate::conda_python("r-miniconda"))
+      }
+    }
+
+    stop(error_msg, call. = FALSE)
+  }
+
+  return(python_path)
+}
+
+configure_virtualenv <- function(env_name, python_path) {
+  SETUPTOOLS_VERSION <- "58.0.0"
+
+  # Create environment if it doesn't exist.
+  if (!reticulate::virtualenv_exists(env_name)) {
+    # Enforce RETICULATE_PYTHON to be the version passed into the function.
+    # This prevents reticulate's miniconda from overriding it, if present.
+    Sys.setenv(RETICULATE_PYTHON = python_path)
+
+    # Check if the virtualenv module is installed. If not, install it.
+    install_virtualenv_module(python_path)
+
+    packageStartupMessage("Creating Python virtual environment...")
+    reticulate::virtualenv_create(
+      envname = env_name,
+      python = python_path,
+      module = "virtualenv",
+      setuptools_version = SETUPTOOLS_VERSION
+    )
+  }
+
+  # Enforce RETICULATE_PYTHON to be the version in the r-reticulate env,
+  # in order to make any changes to setuptools affect the env python
+  # instead of the version on python_path, if it is different.
+  env_python <- reticulate::virtualenv_python("r-reticulate")
+  Sys.setenv(RETICULATE_PYTHON = env_python)
+
+  # Prompt the user to restart R in case the python version changes.
+  # Note: this only seems to work consistently on Mac/Linux, due to
+  # reticulate choosing conda if present over RETICULATE_PYTHON on Windows.
+  loaded_python <- reticulate::py_config()$python
+  if (env_python != loaded_python) {
+    sysname <- Sys.info()[["sysname"]]
+    if (sysname == "Darwin" || sysname == "Linux") {
+      restart_msg <- paste(
+        "The required Python version differs from the pre-loaded Python.",
+        "Restart R and re-run the installation with:",
+        "",
+        "  toolchest::install_toolchest()",
+        "",
+        "Let Toolchest know if this problem persists.",
+        sep = "\n"
+      )
```

Comment:
Did you hit this when installing on the new Mac?

In the same vein, what did the actual install process look like on the new Mac?

Reply:
Yes. `reticulate` needs to load python before it can create a new environment, and it can't "unload" it reliably without restarting the R session, from what I can find. This error is hit when: 
* an existing version of R is loaded
* the `r-reticulate` virtual env does not exist, so reticulate creates it
Any changes that are made (e.g., reinstalling a compatible version of `setuptools`) are only propagated to the loaded version of R. The client is built/run on the `r-reticulate` env, so the install breaks if setuptols needs to be reinstalled.

Reply:
Note that this case only happens at most once (during the first ever package load, when configuring the virtual env and `RETICULATE_PYTHON`).